### PR TITLE
Compatibility with xrootd 5.

### PR DESCRIPTION
**Pull request by ellert**
The old client library libXrdClient is removed.
Also XrdSysDNS is removed.

Do not build the legacy XProofD file server and client (already
disabled by default - give error if someone enables it while building
with xrootd 5).

Do not build the old libNetx.

Build the new libNetNG without XrdSysDNS, use XrdNetAddr instead.

### CODE REVIEW

**Comment by oshadura** [comment]
@phsft-bot build!

**Comment by oshadura** [comment]
@gganis can you please take a look?

File: cmake/modules/SearchInstalledSoftware.cmake [hunk 1]
```diff
@@ -925,6 +925,21 @@ if(xrootd AND XROOTD_VERSIONNUM VERSION_GREATER 300030005)
...
+else()
+  set(netx OFF)
+endif()
+if(xrootd AND XROOTD_VERSIONNUM VERSION_GREATER_EQUAL 500000000)
+  if(xproofd)
+    if(fail-on-missing)
+      message(FATAL_ERROR "XROOTD is version 5 or greater. The legacy xproofd servers can not be built with this version. Use -Dxproofd:BOOL=OFF to disable.")
+    else()
+      message(STATUS "XROOTD is version 5 or greater. The legacy xproofd servers can not be built with this version. Disbaling xproofd' option.")
+      set(xproofd OFF CACHE BOOL "Disabled because xrootd version is 5 or greater" FORCE)
```
Comment (gganis) [inline cmake/modules/SearchInstalledSoftware.cmake:939]: Typo: disbaling -> disabling

**Comment by gganis** [comment]
Xrootd v5 is too young to be set as default, I believe. But the patch can be tested with xrootd v5 as external.

File: cmake/modules/SearchInstalledSoftware.cmake (hunk 1 above)
Comment (oshadura) [reply cmake/modules/SearchInstalledSoftware.cmake:939]: @ellert please take a look..

**Comment by oshadura** [comment]
@Axel-Naumann  I also noticed that clang-tidy is broken here, because we don't build builtin_xrootd target. Should we?

```
Found compiler error(s).
+RESULT_OUTPUT='/home/travis/build/root-project/root/net/netxng/src/TNetXNGSystem.cxx:28:10: error: '\''XrdCl/XrdClFileSystem.hh'\'' file not found [clang-diagnostic-error]
#include <XrdCl/XrdClFileSystem.hh>

         ^'
The command "if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
  .ci/tidy_script.sh
```

File: cmake/modules/SearchInstalledSoftware.cmake (hunk 1 above)
Comment (ellert) [reply cmake/modules/SearchInstalledSoftware.cmake:939]: I have fixed the typo.

**Pull request merged**